File: src/main/scala/fds/scala/queue/Producer.scala
package fds.scala.queue


import io.nats.client.Connection
import io.nats.client.Nats
import java.nio.charset.StandardCharsets


/**
  * Producer stub
  *
  */
object Producer extends App {

  val nc: Connection  = Nats.connect("nats://localhost:4222")

  // Write a start of stream marker
  nc.publish("logs", "Starting Streaming".getBytes(StandardCharsets.UTF_8))

  val fileName = "access.log"
  val logLines = scala.io.Source.fromFile(fileName).getLines

  // TODO use logLines to write each line onto the channel
  for (line <- logLines) {
      // ...
  }

  // TODO write an end of stream marker

}

File: src/main/java/fds/java/queue/Producer.java
package fds.java.queue;


import com.zink.queue.*;

import java.io.FileInputStream;
import java.util.Scanner;


/**
 * Stub for the Producer of information to queue
 *
 * From Fly Docker
 *
 * > docker run -p 4396:4396 zink/fly
 *
 * To find container <run time name>
 * > docker ps
 *
 * To find ipAddress
 * > docker inspect <run time name> | grep -i ipaddress
 *
 */
public class Producer {

    public static void main(String[] args) throws Exception {

        final String ipAddr = "192.168.1.84";
        Connection con = ConnectionFactory.connect(ipAddr);
        WriteChannel wc = con.publish("BBC7");

        wc.write("Hello Consumer");

        /* Read a file from the local file system line by line */
        final String fileName = "file.log";
	    FileInputStream fis = new FileInputStream(fileName);
	    Scanner scanner = new Scanner(fis);
	    while(scanner.hasNextLine()) {
            String line = scanner.nextLine();
            // TODO
            // Write to channel until the file is finished
        }
        // TODO
        // Write an end of stream marker
	}

}
